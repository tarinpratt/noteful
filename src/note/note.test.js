import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import Note from './Note'



describe(`Note component`, () => {
  const props = {
    id: 'x',
    name: 'name',
    modified: new Date(2019, 7, 26),
  }

  it('renders a Note', () => {
    const wrapper = shallow(<Note />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Note w/ props', () => {
    const wrapper = shallow(<Note {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
    <Note />
    </BrowserRouter>, 
  div
  )
  ReactDOM.unmountComponentAtNode(div);
})

})