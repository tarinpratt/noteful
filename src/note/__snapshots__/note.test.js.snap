// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders a Note 1`] = `
<li
  className="noteList"
>
  <h2
    className="noteTitle"
  >
    <Link
      to="/note/undefined"
    />
  </h2>
  <p
    className="noteDate"
  >
    Date modified on  
    Invalid Date
  </p>
  <button
    className="deleteButton"
    onClick={[Function]}
    type="button"
  >
    Delete Note
  </button>
</li>
`;

exports[`Note component renders a Note w/ props 1`] = `
<li
  className="noteList"
  key="x"
>
  <h2
    className="noteTitle"
  >
    <Link
      to="/note/x"
    >
      name
    </Link>
  </h2>
  <p
    className="noteDate"
  >
    Date modified on  
    26th Aug 2019
  </p>
  <button
    className="deleteButton"
    onClick={[Function]}
    type="button"
  >
    Delete Note
  </button>
</li>
`;
